frontend/src/reducers/user.js

```diff
@@ -0,0 +1,165 @@
+          user.actions.setAccessToken({
+            accessToken: json.accessToken,
+          })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        dispatch(user.actions.setName({ name: json.name }));
+      })
+      .catch(err => {
+        dispatch(
+          user.actions.setErrorMessage({ errorMessage: err.toString() })
+        );
+      });
+  };
+};
+
+// Sign up
+export const signUp = (name, email, password) => {
+  return dispatch => {
+    fetch(USERS_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        }
+        throw new Error('Could not create new user');
+      })
+      .then(json => {
+        dispatch(
+          user.actions.setAccessToken({ accessToken: json.accessToken })
+        );
+        dispatch(user.actions.setUserId({ userId: json.userId }));
+        dispatch(user.actions.setName({ name: json.name }));
+        dispatch(user.actions.setErrorMessage({ errorMessage: '' }));
+      })
+      .catch(err => {
+        dispatch(
+          user.actions.setErrorMessage({ errorMessage: err.toString() })
+        );
+      });
+  };
+};
+
+// Secret message
+export const getSecretMessage = () => {
+  return (dispatch, getStore) => {
+    const accessToken = getStore().user.login.accessToken;
+    const userId = getStore().user.login.userId;
+
+    fetch(`${USERS_URL}/${userId}/secret`, {
+      method: 'GET',
+      headers: { Authorization: accessToken },
+    })
+      .then(res => {
+        if (res.ok) {
+          return res.json();
+        }
+        throw new Error(
+          'Could not get requested information. Make sure you are logged in and try again'
+        );
+      })
+      .then(json => {
+        dispatch(user.actions.setSecretMessage({ secretMessage: json }));
+      })
+      .catch(err => {
+        dispatch(
+          user.actions.setErrorMessage({ errorMessage: err.toString() })
+        );
+      });
+  };
+};
+
+// Logout
+export const logout = () => {
+  return (dispatch, getStore) => {
+    const accessToken = getStore().user.login.accessToken;
+    fetch(`${USERS_URL}/logout`, {
+      method: 'POST',
+      headers: { Authorization: accessToken },
+    })
+      .then(res => {
+        if (!res.ok) {
+          throw new Error('Failed to logout');
+        }
+        return res.json();
+      })
+      .catch(err => {
+        dispatch(
+          user.actions.setErrorMessage({ errorMessage: err.toString() })
+        );
+      });
+    dispatch(user.actions.setName({ name: '' }));
+    dispatch(user.actions.setUserId({ userId: 0 }));
+    dispatch(user.actions.setSecretMessage({ secretMessage: '' }));
+    dispatch(user.actions.setErrorMessage({ errorMessage: '' }));
+    dispatch(user.actions.setAccessToken({ accessToken: null }));
+    localStorage.clear();
```

Comment:
In bigger apps, you can want to store more data in the local storage (dark/bright mode etc). Cleaning it all in a single function is not a good practice. You should always remove elements one by one to prevent losing unrelated to this functionality fields from browser memory. 🛑

Reply:
I have fixed this now. Thanks for input :)